### ISSUE TITLE
vmware-iso builder can't connect to VNC

### ISSUE THREAD

**Issue by andrew-regan** [hashicorp/packer#4798]
Build fails with:  _ui error: ==> vmware-iso: Error connecting to VNC: dial tcp 10.10.10.211:5901: getsockopt: connection refused_

**Packer Version:** Issue exists in v1.0.0 and v.0.12.3. Build works w/o issue or changes at v0.12.2.
**Host platform:** Docker container sourced from centos:centos7
**Debug Output:** https://gist.github.com/andrew-regan/1f1f0b18f50d4acc3a5cb2cc862a937d
**Packer.json:** https://gist.github.com/andrew-regan/0d481f6db8bc1045a62101a6e02fbefb

**Comment by mwhooker** [comment]
make sure you have network access to that host and port. Try running packer with `-on-error=ask` to have it pause when it encounters an errors. Then try to connect to VNC manually. If you can connect then there's likely an issue with packer. If not, please investigate your network settings.

**Comment by andrew-regan** [comment]
It works, but I'll double check again with `-on-error=ask`. This build is fully automated and working w/o issues with Packer v0.12.2. When I change my Dockerfile to point at v0.12.3 or v1.0.0 it fails.

**Comment by mwhooker** [comment]
ahh, sorry, I missed that it was working in `v0.12.2`.

I don't see any commits between 12.2 and 12.3 that might introduce this, but I'll keep looking. If you were able to run a git bisect to identify that bad commit, that would be a huge help.

**Comment by mwhooker** [comment]
The only thing I can think of would be 29dd20da6856fa21d71be2e20e433ce51eaf2714

Would you be able to test with a binary with that commit reverted?

**Comment by andrew-regan** [comment]
Yes.

**Comment by andrew-regan** [comment]
v0.12.3 output that reproduces the same issue: https://gist.github.com/andrew-regan/43f05887082a5d8aa54a4722171d5d6d
v0.12.2 output that is past the issue: https://gist.github.com/andrew-regan/5164bc1a55d32d316425396c21bd571d

The only obvious difference is that 0.12.2 goes into a loop searching for a VNC port while v0.12.3 and v1.0.0 don't:

```
2017/04/18 00:38:16 packer: 2017/04/18 00:38:16 Trying address: 10.10.10.211:5900...
2017/04/18 00:38:17 packer: 2017/04/18 00:38:17 Timeout connecting to: 10.10.10.211:5900 (check firewall rules)
2017/04/18 00:38:17 packer: 2017/04/18 00:38:17 Trying address: 10.10.10.211:5901...
2017/04/18 00:38:18 packer: 2017/04/18 00:38:18 Timeout connecting to: 10.10.10.211:5901 (check firewall rules)
2017/04/18 00:38:18 packer: 2017/04/18 00:38:18 Trying address: 10.10.10.211:5902...
...
```

**Comment by mwhooker** [comment]
ok I think this is the PR that introduced the problem for you https://github.com/hashicorp/packer/pull/4610

it changes the timeout from 5 seconds to 30 seconds while looking for a VNC port. (https://github.com/hashicorp/packer/pull/4480 changed from 1 to 5 seconds). I wonder if the added time introduces a race condition. Are there other packer processes trying to build on the same esxi host?

I think probably the proper solution is to make this configurable, as suggested in #3205

**Comment by andrew-regan** [comment]
Yes. I ran all 3 tests against the same host serially.

**Comment by andrew-regan** [comment]
Sorry...been swamped and haven't had a chance to get back to this yet. Still on my ToDo list...

**Comment by mwhooker** [comment]
@andrew-regan np. I still have to make a PR to fix this. I'll let you know when I have something to test

**Comment by mwhooker** [comment]
@andrew-regan I think I fixed this in #4919

Would you mind taking a look if you get a chance? Let me know if you need help testing
